homeassistant/components/input_button/__init__.py

```diff
@@ -0,0 +1,171 @@
+"""Support to keep track of user controlled buttons for within automations."""
+from __future__ import annotations
+
+import logging
+from typing import cast
+
+import voluptuous as vol
+
+from homeassistant.components.button import SERVICE_PRESS, ButtonEntity
+from homeassistant.const import (
+    ATTR_EDITABLE,
+    CONF_ICON,
+    CONF_ID,
+    CONF_NAME,
+    SERVICE_RELOAD,
+)
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+from homeassistant.helpers import collection
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.helpers.restore_state import RestoreEntity
+import homeassistant.helpers.service
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.typing import ConfigType
+
+DOMAIN = "input_button"
+
+_LOGGER = logging.getLogger(__name__)
+
+CREATE_FIELDS = {
+    vol.Required(CONF_NAME): vol.All(str, vol.Length(min=1)),
+    vol.Optional(CONF_ICON): cv.icon,
+}
+
+UPDATE_FIELDS = {
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_ICON): cv.icon,
+}
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: cv.schema_with_slug_keys(vol.Any(UPDATE_FIELDS, None))},
+    extra=vol.ALLOW_EXTRA,
+)
+
+RELOAD_SERVICE_SCHEMA = vol.Schema({})
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+
+
+class InputButtonStorageCollection(collection.StorageCollection):
+    """Input button collection stored in storage."""
+
+    CREATE_SCHEMA = vol.Schema(CREATE_FIELDS)
+    UPDATE_SCHEMA = vol.Schema(UPDATE_FIELDS)
+
+    async def _process_create_data(self, data: dict) -> vol.Schema:
+        """Validate the config is valid."""
+        return self.CREATE_SCHEMA(data)
```

Comment:
It's weird we don't have coverage here, is codecov in the wrong or is something missing from the tests?

Reply:
Hmm good question, will take a look.